src/storage/mod.rs

```diff
@@ -203,6 +317,20 @@ where
         }
     }
 
+    /// Returns an entry to the component associated to the entity.
+    ///
+    /// Note: This does not immediately error if the entity is dead.
```

Comment:
I think we should use `Result` here.

Reply:
Are we sure? That would make this require a bit more boilerplate for this stuff then:
```rust
if let Ok(entry) = storage.entry(entity) {
    entry.or_insert(component);
}
```
At least until we get `try` to work outside of just functions.

Reply:
And now the boilerplate is at another location
```
if let Some(component) = storage.entry(entity).or_insert(...) {
    ...
}
```

Reply:
I'm somewhat worried about enforcing that `Entry` is only created with a live `Entity`, is there a case where an entity could be deleted after you create an entry?

Reply:
In another thread. But it is not visible until maintenance.

Reply:
Shrödinger entity

Reply:
It can be killed atomically. But atomically killed enities are considered alive until maintenance.

Reply:
True, I messed around with it and couldn't find an easy way to kill the entity while having an entry, so I guess it should be fine.